### DISCUSSION TITLE
Inconsistent Results with ProbeLocation Over Time

### DISCUSSION THREAD

**Discussion by evdkmv** [Kitware/trame#576] [Trame/ParaView] [answered]
Hello,

I am trying to extract the solution over time steps at a particular point using the [ProbeLocation](https://discourse.paraview.org/t/paraview-foam-reader-decomposed-case/12157/8) filter. The code works correctly when I change self.state.probe_location for the first time. However, when I change self.state.probe_location again, the solution_values array returns only [0.0, 0.0, 0.0, ...].

The change in self.state.probe_location is detected correctly, and the time steps remain consistent. It seems like some pipeline update is missing or not being properly triggered. 

Does anyone know what I might be missing or how to ensure the ProbeLocation filter updates correctly on subsequent changes of the state variable?

Any help or suggestions would be greatly appreciated!

Thank you.

```
    @change("probe_location")
    def extract_value_at_location_over_time(self, **kwargs):

        if self.state.probe_location is None:
            return

        time_steps = self.time_keeper.TimestepValues

        probeLocation1 = simple.ProbeLocation(
            Input=self.active_source,
            ProbeType="Fixed Radius Point Source",
        )
        probeLocation1.ProbeType.Center = self.state.probe_location

        simple.Show()

        solution_values = []

        for time in time_steps:
            self.animation_scene.AnimationTime = time

            probe_data = simple.servermanager.Fetch(probeLocation1)
            point_data = probe_data.GetPointData()
            u_value = point_data.GetArray(self.state.scalar_field)
            value = u_value.GetTuple(0)[0]
            solution_values.append(value)

        print(f"solution_values: {solution_values}")
```

**Comment by jourdain** [comment, marked as answer]
That would be a ParaView question on discourse. But I see a couple of possible issue.

1. Why do you create a ProbeLocation every time?
2. Why do you create a representation?
3. If you re-use the ProbeLocation, you could just update the Center and call UpdatePipeline on it

**Reply by evdkmv** [reply]
Yes, it works. Thanks Sebastien!

Extracting the probe_location_filter already solved the problem. Substituting show() with UpdatePipeline() significantly increased the speed.

Here's the code for reference:

```
class Viewer:
    def __init__(self, server=None):
        # ... other initialization code ...
        self.probe_location_filter = None

    def create_probe_location_filter(self):
        if self.probe_location_filter is None:
            self.probe_location_filter = simple.ProbeLocation(
                Input=self.active_source,
                ProbeType="Fixed Radius Point Source",
            )

    @change("probe_location")
    def extract_value_at_location_over_time(self, **kwargs):

        if self.state.probe_location is None:
            return

        time_steps = self.time_keeper.TimestepValues
        self.create_probe_location_filter()
        self.probe_location_filter.ProbeType.Center = self.state.probe_location

        solution_values = []

        for time in time_steps:
            self.probe_location_filter.UpdatePipeline(time=time)
            probe_data = simple.servermanager.Fetch(self.probe_location_filter)
            point_data = probe_data.GetPointData()
            u_value = point_data.GetArray(self.state.scalar_field)
            value = u_value.GetTuple(0)[0]
            solution_values.append(value)

        print(f"solution_values: {solution_values}")
```
